pkg/ottl/ottlfuncs/func_is_match.go

```diff
@@ -38,6 +38,9 @@ func IsMatch[K any](target ottl.Getter[K], pattern string) (ottl.ExprFunc[K], er
 		if err != nil {
 			return nil, err
 		}
+		if val == nil {
+			return false, nil
+		}
 
 		switch v := val.(type) {
```

Comment:
Not related to this PR, but keep in mind that this path is a slippery path, and will for sure hurt us. The fact that you accept multiple types instead of asking users to call this as IsMatch(..., string(value)) and enforce always having string here. The reason is because I feel like this will get duplicate in multiple places and will diverge.

You can at least start by adding "String()" func and call it on the value to avoid duplicate code.

Reply:
I don't think we want to force users to use a String() Converter function every time they use IsMatch so I like that the function handles conversion for you. 

If your concern is only the potential duplication of the type switch statement for strings, I can pull that out into a helper function for future reuse.

Reply:
Let's start with avoiding potential duplication, I bet we already duplicate this.